modules/core/src/smithy4s/schema/CompilationCache.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+package schema
+import scala.collection.mutable.{Map => MMap}
+
+trait CompilationCache[F[_]] {
```

Comment:
The construct that actually abstract over the mutable variable we use for caching. Can be implemented as a no-op, which is useful to retain the API sane.